### PR TITLE
perf: disable compression for WAL

### PR DESCRIPTION
**Pull request by NamanJain8**
This PR disables compression from WAL in badger. WAL is supposed to be small and fast, Cache/compression makes it a little slow (we save some disk though but its not worth the slowness).
Cache has not been disabled as `w` directory goes to about 30GB as found in 1TB write benchmark for dgraph.

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6255)

 
 

Docs Preview: [Dgraph Preview](https://dgraph-215387230d-88874.surge.sh)

### CODE REVIEW

File: worker/server_state.go [hunk 1]
```diff
@@ -128,9 +128,8 @@ func (s *ServerState) initStorage() {
 		// Write Ahead Log directory
 		x.Checkf(os.MkdirAll(Config.WALDir, 0700), "Error while creating WAL dir.")
 		opt := badger.LSMOnlyOptions(Config.WALDir)
```
Comment (NamanJain8) [inline worker/server_state.go:130]: We are relying on the default badger options for disabling. Is it okay? Or an explicit disable should be done?

**Review by jarifibrahim — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6255#-:-MFUiRbk2WnE2bq3Z0ah:bnfp4nl)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @jarifibrahim, @manishrjain, @NamanJain8, and @vvbalaji-dgraph)

___
*[worker/server_state.go, line 130 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/6255#-MFF0ka5CKF76l7r3lBV-r1-130:-MFUfjDv0qMKGPvtKoci:bkgp01j) ([raw file](https://github.com/dgraph-io/dgraph/blob/b59b4b447b530971921e18efdc00687244db0d5c/worker/server_state.go#L130)):*
<details><summary><i>Previously, NamanJain8 (Naman Jain) wrote…</i></summary><blockquote>

We are relying on the default badger options for disabling as caching/compression are disabled by default. Is it okay? Or an explicit disable should be done?
</blockquote></details>

I think that's okay.
___
*[worker/server_state.go, line 85 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/6255#-MFUg77t1wG2j4vWrcXc:-MFUg77t1wG2j4vWrcXd:b-4ayr6d) ([raw file](https://github.com/dgraph-io/dgraph/blob/5f4cbb3ca563d0f178bbec8870e6eb9586f7728c/worker/server_state.go#L85)):*
> ```Go
> 		// Settings for the write-ahead log.
> 		badgerTables = Config.BadgerWalTables
> 		badgerVlog = Config.BadgerWalVlog
> ```

We should explicitly disable the cache and compression here so that there is no confusion.

**Review by ashish-goswami — CHANGES_REQUESTED** [review]
:lgtm: 

> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6255#-:-MFVID7H5jeEMB8rBUGR:bg7igkv)* status: all files reviewed, 3 unresolved discussions (waiting on @manishrjain, @NamanJain8, and @vvbalaji-dgraph)

___
*[worker/server_state.go, line 86 at r3](https://reviewable.io/reviews/dgraph-io/dgraph/6255#-MFVI1Nx7sb22VgMMfY_:-MFVI1Ny3a9xLYUV0U4i:b9pu11j) ([raw file](https://github.com/dgraph-io/dgraph/blob/0af1e0ffaa57ae832ca4da3e5df05afa006fa2b9/worker/server_state.go#L86)):*
> ```Go
> 		badgerTables = Config.BadgerWalTables
> 		badgerVlog = Config.BadgerWalVlog
> 		// disable cache and compression explicitly for write-ahead log
> ```

Please also write in comment reason for doing this.

**Review by ashish-goswami — COMMENTED** [review]
:lgtm: 

> Reviewed 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6255#-:-MFjDsNM0Mi830OWZ_6D:bg7igkv)* status: all files reviewed, 3 unresolved discussions (waiting on @manishrjain, @NamanJain8, and @vvbalaji-dgraph)

File: worker/server_state.go [hunk 2]
```diff
@@ -77,25 +77,27 @@ func setBadgerOptions(opt badger.Options, wal bool) badger.Options {
...
 		badgerTables = Config.BadgerWalTables
 		badgerVlog = Config.BadgerWalVlog
+		// Disable compression for WAL as it is supposed to be fast. Compression makes it a
+		// little slow (Though we save some disk space but it is not worth the slowness).
+		opt.Compression = options.None
 	} else {
 		// Settings for the data directory.
 		badgerTables = Config.BadgerTables
 		badgerVlog = Config.BadgerVlog
+		glog.Infof("Setting Badger Compression Level: %d", Config.BadgerCompressionLevel)
```
Comment (parasssh) [inline worker/server_state.go:93]: this is more than what the PR description. Do we need this in this PR.

Comment (NamanJain8) [reply worker/server_state.go:93]: This is just a refactor. As we have `if-else` block for pstore/wstore, so having the compression settings set within that block makes more sense I think.
So which earlier used to be a common code for p/w, now is separate for both.

Comment (parasssh) [reply worker/server_state.go:93]: No. I meant that earlier there was no logic but now you are looking at the CompressionLevel and setting some values in badger options.

**Pull request merged**